packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -594,6 +605,23 @@ export class Table extends TableBase {
     this.billingMode = props.billingMode || BillingMode.PROVISIONED;
     this.validateProvisioning(props);
 
+    let streamSpecification: CfnTable.StreamSpecificationProperty | undefined;
+    if (props.replicationRegions) {
+      if (props.stream && props.stream !== StreamViewType.NEW_AND_OLD_IMAGES) {
+        throw new Error('`stream` must be set to `NEW_AND_OLD_IMAGES` when specifying `replicationRegions`');
+      }
+      streamSpecification = { streamViewType: StreamViewType.NEW_AND_OLD_IMAGES };
+
+      if (props.billingMode && props.billingMode !== BillingMode.PAY_PER_REQUEST) {
+        throw new Error('The `PAY_PER_REQUEST` billing mode must be used when specifying `replicationRegions`');
+      }
+      this.billingMode = BillingMode.PAY_PER_REQUEST;
+    } else if (props.stream) {
+      streamSpecification = { streamViewType : props.stream };
+    } else {
+      streamSpecification = undefined;
```

Comment:
This fall through is probably not needed since the value will be `undefined` if none of the conditions are met